src/yunohost/hook.py

```diff
@@ -288,10 +289,8 @@ def hook_callback(action, hooks=[], args=None, no_trace=False, chdir=None,
             else:
                 post_callback(name=name, priority=priority, path=path,
                               succeed=True)
-            try:
-                result[state][name].append(path)
-            except KeyError:
-                result[state][name] = [path]
+
+            result[name] = {'path' : path, 'state' : state, 'stdreturn' : hook_return }
```

Comment:
This PR doesn't change a lot of code using the 'state' key, like here https://github.com/YunoHost/yunohost/blob/115b557b67893111d5b7708f6b3b3226615f3092/src/yunohost/backup.py#L613

The easiest is to keep the result[state][name] like that so something like:
```
result[state][name] = {'path' : path, 'state' : state, 'stdreturn' : hook_return }
```

Reply:
I think I changed that because it was a request from @alexAubin . But not sure.

Can @alexAubin  confirm this ?

Reply:
Uh yeah I don't understand the trouble here ?

To me it doesn't make sense to just put things in either the "succeed" or "failed" box instead of just having directly a dictionnary of hook name (`result[name]`) ?

Dunno why that troublesome to do this instead of keeping the existing `result[state][name]` ?

Reply:
To do this, you need to change a lot of things and you can't just make:
```
ret = hook_callback(...)
if ret['failed']:
    # Some error management
```

This kind of pattern is used in a lot of piece of code. We can't change the format of the return without check all hook_callback call...

Reply:
In more, previously there was possibility to get several hook path for the same name. I think it could be several path on setup using custom hook. I believe mailman_ynh use this feature.

Reply:
Well,

So what should I do for that now ?

Reply:
I think we did check all the `hook_callback` call, though I would need to double-check ... but it's not that much a big deal imho :/

About the "several path" thing, gotta check what is the exact behavior but that doesn't look like a blocker thing to me :/

Reply:
Now it should be fixed